op-challenger/game/fault/register.go

```diff
@@ -38,16 +40,34 @@ func RegisterGameTypes(
 	cfg *config.Config,
 	txMgr txmgr.TxManager,
 	client *ethclient.Client,
-) {
+) (CloseFunc, error) {
+	var closer CloseFunc
+	createL2Client := func() (cannon.L2HeaderSource, error) {
+		l2Client, err := ethclient.DialContext(ctx, cfg.CannonL2)
+		if err != nil {
+			return nil, fmt.Errorf("dial l2 client %v: %w", cfg.CannonL2, err)
+		}
+		closer = l2Client.Close
+		return l2Client, nil
+	}
 	if cfg.TraceTypeEnabled(config.TraceTypeOutputCannon) {
-		registerOutputCannon(registry, ctx, logger, m, cfg, txMgr, client)
+		l2Client, err := createL2Client()
+		if err != nil {
+			return nil, err
+		}
+		registerOutputCannon(registry, ctx, logger, m, cfg, txMgr, client, l2Client)
 	}
 	if cfg.TraceTypeEnabled(config.TraceTypeCannon) {
-		registerCannon(registry, ctx, logger, m, cfg, txMgr, client)
+		l2Client, err := createL2Client()
```

Comment:
ugh, that was dumb.  I meant for that method to ensure it only created one.

Reply:
Wound up just simplifying it so that we have a separate if to decide if the client is required. There's a chance we'll add a new trace type in the future that needs the client and forget to update the if, but then challenger will panic on start up as soon as its enabled so it won't go unnoticed.  And I think the resulting code is simpler.